### PR TITLE
Add pso ShaderCreationError

### PR DESCRIPTION
**Pull request by Gordon-F**
As discussed in the matrix. Add new pso creation error - ShaderCreationError.

PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends: GL (CI should check all other errors)

### CODE REVIEW

File: src/hal/src/pso/mod.rs [hunk 1]
```diff
@@ -21,6 +21,9 @@ pub enum CreationError {
     /// Unknown other error.
     #[error("Implementation specific error occurred")]
     Other,
+    /// Shader module creation error.
+    #[error("Shader creation failed: {0:}")]
+    ShaderCreationError(String),
```
Comment (kvark) [inline src/hal/src/pso/mod.rs:26]: I think we always know the shader stage when we issue this error, don't we? In this case, we should make it `ShaderCreationError(Stage, String)`. Would you be open to try it out?

**Review by kvark — COMMENTED** [review]
Thank you for doing this! I have one concern to address

**Comment by Gordon-F** [comment]
CI is finally happy. Should be ready to merge.

**Review by kvark — APPROVED** [review]
superb, thank you for addressing this!
bors r+

**Pull request merged**